src/hooks/usePrevious.ts

```diff
@@ -0,0 +1,25 @@
+// Copyright 2021 Parity Technologies (UK) Ltd.
+// This file is part of Parity Bridges UI.
+//
+// Parity Bridges UI is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Parity Bridges UI is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Parity Bridges UI.  If not, see <http://www.gnu.org/licenses/>.
+
+import { useRef, useEffect } from 'react';
+
+export default function usePrevious(value: any) {
+  const ref = useRef();
+  useEffect(() => {
+    ref.current = value;
+  });
+  return ref.current;
+}
```

Comment:
This thing looks super hacky, why do we `useRef`? Can't we have an explicit state with previous value (cause I believe that's what this thing is supposed to do).

Reply:
Actually this hook is the best way to retrieve the previous value without having a second state to keep the value (plus its reusable for multiple components - so I would leave it as is.

Reply:
yes, actually this is quite used and there is a lot of  examples out there.